### ISSUE TITLE
Settings and widgets should change concurrently

### ISSUE THREAD

**Issue by tybug** [circleguard/circleguard#40]
We have to do a lot of messiness with slots if we want this functionality currently. Not sure what this would look like but the current idea (with very little investigation to back it up) is to have a single slot that gets emitted with (setting name, new value) whenever a setting is written to. Widgets can then subscribe to this signal by subclassing a `Linked` class or so, and implement on_setting_changed. The single slot would connect to a `Linked` class attribute, and on_setting_changed would only be called if the slot setting_name matches the filter given at instantiation time.

The opposite of this - updating settings using an equivalent signal on a widget - could probably also be implemented here, but I'm not so sure of the details of that one. Will have to see if it's feasible

tldr abstract by adding a way to subscribe to an on_setting_changed signal
